src/cart/cart.controller.ts

```diff
@@ -0,0 +1,116 @@
+import express, { Express, Handler, Router, NextFunction, Request, Response } from 'express';
+import Joi from 'joi';
+import { CartBase, ProductBase, UpdateCartRequestBody } from '../types';
+import { CartService } from './cart.service';
+import { calculateTotal } from '../utils/utils';
+
+const CartController = (cartRepository: CartBase, productRepository: ProductBase) : Router => {
+  const cartRouter: Router = express.Router();
+  const cartService: CartService = new CartService(cartRepository);
+
+  const putCartSchema = Joi.object({
+    productId: Joi.string().uuid().required(),
+    count: Joi.number().integer().min(0).required()
+  });
+
+  const validatePutCartBody = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      console.log(req.body);
+      await putCartSchema.validateAsync(req.body);
+      next();
+    } catch (err) {
+      console.log("logging there", err)
+      res.status(400).send({
+        "data": null,
+        "error": {
+          "message": "Products are not valid"
+        }
+      });
+    }
+  }
+
+  cartRouter.get('/cart', (req, res, next) => {
+    const userId = req.get('x-user-id');
+    const cart = cartService.getUserCart(userId as string);
+    res
+      .status(200)
+      .send({
+        "data": {
+          "cart": {
+            "cart": cart.id,
```

Comment:
M: Change property name
```suggestion
            "id": cart.id,
```